### PR TITLE
issue #89. Use importlib.metadata rather than pkg_resources

### LINKED ISSUE #89: Use importlib.metadata rather than pkg_resources
### What is your issue?

We should update https://github.com/xarray-contrib/xbatcher/blob/main/xbatcher/__init__.py to use importlib.metadata rather than pkg_resources for accessing version information (ref. https://setuptools.pypa.io/en/latest/pkg_resources.html)

### PR DESCRIPTION
**Pull request by norlandrhagen**
**Description of proposed changes**

PR for Issue #89 

Fixes #89 

cc @maxrjones

### CODE REVIEW

**Review by maxrjones — COMMENTED** [review]
Thanks @norlandrhagen!

What's the motivation for placing the code in a new module rather than ``__init__.py``?

**Comment by norlandrhagen** [comment]
No strong motivation for separating out version from `__init__`. Just decided to mirror [Pandas](https://github.com/pandas-dev/pandas/blob/main/pandas/_version.py) and cmip6. Will gladly move it back into `__init__.py` if you think that's best.

**Comment by maxrjones** [comment]
> No strong motivation for separating out version from `__init__`. Just decided to mirror [Pandas](https://github.com/pandas-dev/pandas/blob/main/pandas/_version.py) and cmip6. Will gladly move it back into `__init__.py` if you think that's best.

Thanks for the explanation! I think since this is a small amount of code relative to those examples we could minimize the number of modules and keep it in `__init__`.  After that it'll be great to merge this in!

File: xbatcher/_version.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from importlib.metadata import PackageNotFoundError, version
+
+try:
+    __version__ = version(__name__)
+except PackageNotFoundError:
+    # package is not installed
+    __version__ = 'unknown'
```
Comment (weiji14) [inline xbatcher/_version.py:7]: Need to delete this file after 25595fbe13902e343734705d6c10bab9fa152285

Comment (norlandrhagen) [reply xbatcher/_version.py]: Thanks for the catch! @weiji14

**Review by weiji14 — APPROVED** [review]
Not too familiar with pkg_resources/importlib, but looks ok to me!

File: xbatcher/__init__.py [hunk 2]
```diff
@@ -1,11 +1,11 @@
-from pkg_resources import DistributionNotFound, get_distribution
+from importlib.metadata import PackageNotFoundError, version
```
Comment (maxrjones) [inline xbatcher/__init__.py:1]:
```suggestion
from importlib.metadata import (
    PackageNotFoundError as _PackageNotFoundError,
    version as _version,
)
```

File: xbatcher/__init__.py [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
 
 from .accessors import BatchAccessor  # noqa: F401
 from .generators import BatchGenerator  # noqa: F401
 from .util.print_versions import show_versions  # noqa: F401
 
 try:
-    __version__ = get_distribution(__name__).version
-except DistributionNotFound:  # noqa: F401; pragma: no cover
+    __version__ = version(__name__)
+except PackageNotFoundError:
```
Comment (maxrjones) [inline xbatcher/__init__.py:9]:
```suggestion
    __version__ = _version(__name__)
except _PackageNotFoundError:
```

**Review by maxrjones — COMMENTED** [review]
A couple suggestions so that the metadata imports aren't included in xbatcher's list of public objects.

**Review by maxrjones — APPROVED** [review]
Thanks @norlandrhagen!

**Pull request merged**